infrasim/model.py

```diff
@@ -1714,183 +1721,34 @@ def precheck(self):
-
-                path_chassiscontrol = dst
-                bmc_obj.set_chassiscontrol_script(path_chassiscontrol)
-
-            if not has_option(self.__node, "bmc", "lancontrol"):
-                os.symlink(os.path.join(config.infrasim_template,
-                                        "lancontrol"),
-                           os.path.join(self.workspace,
-                                        "script",
-                                        "lancontrol"))
-
-                path_lancontrol = os.path.join(self.workspace,
-                                               "script",
-                                               "lancontrol")
-                bmc_obj.set_lancontrol_script(path_lancontrol)
-
-            # Render connection port/device
-            if has_option(self.__node, "type"):
-                bmc_obj.set_type(self.__node["type"])
-
-            if has_option(self.__node, "sol_device"):
-                bmc_obj.set_sol_device(self.__node["sol_device"])
-
-            if has_option(self.__node, "ipmi_console_port"):
-                bmc_obj.set_port_ipmi_console(self.__node["ipmi_console_port"])
-
-            if has_option(self.__node, "bmc_connection_port"):
-                bmc_obj.set_port_qemu_ipmi(self.__node["bmc_connection_port"])
-
-            bmc_obj.set_workspace(self.workspace)
-            bmc_obj.enable_sol(self.__sol_enabled)
-            bmc_obj.init()
-            bmc_obj.write_bmc_config(os.path.join(self.workspace,
-                                                  "data",
-                                                  "vbmc.conf"))
-
-        # VI. Move emulation data
-        # Update identifier accordingly
-        path_emu_dst = os.path.join(self.workspace, "data")
-        if has_option(self.__node, "bmc", "emu_file"):
-            shutil.copy(self.__node["bmc"]["emu_file"], path_emu_dst)
-        else:
-            node_type = self.__node["type"]
-            path_emu_src = os.path.join(config.infrasim_data, "{0}/{0}.emu".format(node_type))
-            shutil.copy(path_emu_src, os.path.join(path_emu_dst, "{}.emu".
-                                                   format(node_type)))
-
-        # VII. Move bios.bin
-        path_bios_dst = os.path.join(self.workspace, "data")
-        if has_option(self.__node, "compute", "smbios"):
-            shutil.copy(self.__node["compute"]["smbios"], path_bios_dst)
-        else:
-            node_type = self.__node["type"]
-            path_bios_src = os.path.join(config.infrasim_data,
-                                         "{0}/{0}_smbios.bin".format(node_type))
-            shutil.copy(path_bios_src, os.path.join(path_emu_dst,
-                                                    "{}_smbios.bin".
-                                                    format(node_type)))
-        # Place holder to sync serial number
+        self.workspace = Workspace()
+        self.workspace.set_node_info(self.__node)
+        if not self.__is_running():
```

Comment:
if failed stop a node, then we run an external command kill/pkill to force the node stopped, started over again, the workspace will not be updated.